### PR TITLE
[matomo-tracker-react-native] - Add type definitions

### PR DESCRIPTION
**Pull request by zecergin**
Please fill in this template.

- [+] Use a meaningful title for the pull request. Include the name of the package modified.
- [+] Test the change in your own code. (Compile and run.)
- [+] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [+] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [+] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [+] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [+] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [+] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [+] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [+] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [+] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [+] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by gabritto — APPROVED** [review]
Thanks, this looks very good. One minor thing though: there seems to be a variable named `MatomoContext` missing from `index.d.ts`. Please consider adding it before merging. :)

File: types/matomo-tracker-react-native/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// Type definitions for matomo-tracker-react-native 0.3
+// Project: https://github.com/donni106/matomo-tracker-react-native
+// Definitions by: Zeynep Ece Ergin <https://github.com/zecergin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.4
```
Comment (peterblazejewicz) [inline types/matomo-tracker-react-native/index.d.ts:5]: why this version? (which was just recently the 'top notch' version) what language feature depends on it here?

File: types/matomo-tracker-react-native/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export interface Download {
+  download: string;
+  url?: string;
+  userInfo?: {
+    uid?: string
+  };
+}
+
+export function useMatomo(): {
+  trackAppStart: (params: AppStart) => undefined | Promise<Response>,
```
Comment (peterblazejewicz) [inline types/matomo-tracker-react-native/index.d.ts:59]: just a comment, the default is other way around, eg..g when using VScode + Copilot, or more common sense

File: types/matomo-tracker-react-native/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+export interface MatomoProviderProps {
+  instance: MatomoTracker;
+  children: React.ReactElement;
+}
+
+export function MatomoProvider(props: MatomoProviderProps): JSX.Element;
+
+export const MatomoContext: React.Context<{}>;
+export interface InstanceProps {
```
Comment (peterblazejewicz) [inline types/matomo-tracker-react-native/index.d.ts:76]:
```suggestion

export interface InstanceProps {
```
space, be consistent

File: types/matomo-tracker-react-native/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    uid?: string
+  };
+}
+
+export interface Action {
+  name: string;
+  userInfo?:
+  {
+    uid?: string
+  };
```
Comment (peterblazejewicz) [inline types/matomo-tracker-react-native/index.d.ts:20]:
```suggestion
  userInfo?:
  {
    uid?: string
  };
```
this block is recurring theme

**Review by gabritto — APPROVED** [review]

**Comment by zecergin** [comment]
Ready to merge

**Pull request merged**